File: check_cache.py
import json
d = json.load(open('/home/ubuntu/trading-bot/cache/banknifty_5yr.json'))
print('type:', type(d).__name__)
print('is list:', isinstance(d, list))
if isinstance(d, dict):
    print('keys:', list(d.keys())[:5])
    # Try to find the candles array
    for k, v in d.items():
        if isinstance(v, list) and len(v) > 100:
            print(f'  {k}: list of {len(v)}, sample:', v[0] if v else None)
elif isinstance(d, list):
    print('len:', len(d))
    print('sample[0]:', d[0] if d else None)

File: check_cache.js
const fs = require('fs');
const d = JSON.parse(fs.readFileSync('cache/banknifty_5yr.json', 'utf-8'));
console.log('type:', typeof d, 'isArray:', Array.isArray(d));
if (!Array.isArray(d)) {
  console.log('keys:', Object.keys(d).slice(0, 5));
  const first = Object.values(d)[0];
  console.log('first val sample:', JSON.stringify(first).slice(0, 200));
} else {
  console.log('length:', d.length, 'sample:', JSON.stringify(d[0]));
}
